lib/mix/lib/mix/tasks/compile.ex

```diff
@@ -91,10 +91,16 @@ defmodule Mix.Tasks.Compile do
     load_erl_config(opts)
 
     {res, diagnostics} =
-      Mix.Task.run("compile.all", args)
-      |> List.wrap()
-      |> Enum.map(&Mix.Task.Compiler.normalize(&1, :all))
-      |> Enum.reduce(&merge_diagnostics/2)
```

Comment:
The simplest fix here would be to replace `Enum.reduce(&merge_diagnostics/2)` by `Enum.reduce({:ok, []}, &merge_diagnostics/2)`. Then we just ensure that calling the umbrella app does not raise.

Reply:
Something like that?

```
diagnostics =
      Mix.Task.run("compile.all", args) |> List.wrap()
      |> Enum.map(&Mix.Task.Compiler.normalize(&1, :all))

    {res, diagnostics} =
      case diagnostics do
        [] ->
          Enum.reduce(diagnostics, {:error, []}, &merge_diagnostics/2)

        _ ->
          Enum.reduce(diagnostics, &merge_diagnostics/2)
      end
```